src/base/OpenMCCellAverageProblem.C

```diff
@@ -1835,39 +1872,87 @@ OpenMCCellAverageProblem::meshFilter()
   {
     const auto & translation = _mesh_translations[i];
     auto meshFilter = dynamic_cast<openmc::MeshFilter *>(openmc::Filter::create("mesh"));
-    meshFilter->set_mesh(mesh_index);
+    meshFilter->set_mesh(_mesh_index);
     meshFilter->set_translation({translation(0), translation(1), translation(2)});
     mesh_filters.push_back(meshFilter);
   }
 
   return mesh_filters;
 }
 
+void
+OpenMCCellAverageProblem::resetTallies()
+{
+  if (_needs_global_tally)
+  {
+    // erase tally
+    auto idx = openmc::model::tallies.begin() + _global_tally_index;
+    openmc::model::tallies.erase(idx);
```

Comment:
We'll want to remove entries in the `model::tally_map` as well.

Reply:
Shouldn't that quantity stay the same? If I'm going to re-init the same tally, just with potentially a different filter?

Reply:
That's a map of from tally ID to the tally's index into the `openmc::model::tallies` vector. You may be right that Cardinal will generally replace all of the entires in this map and there won't be any invalid entires, but as a matter of good practice I'd recommend removing the entries for the removed tallies from this map as well to protect from a potential bug in the future if we change how tallies are handled for some reason.

Reply:
Actually, does this code not already handle the `tally_map` removal? Once I delete the unique ptr, and it goes out of scope?

```
Tally::~Tally()
{
  model::tally_map.erase(id_);
}
```

Reply:
Oh yeah you're totally right! Thanks for spotting that. In fact, I'm now noticing that there are other classes that would benefit from this line in their constructor.

Onward!